### PR TITLE
HBASE-22903 : Table to RegionStatesCount metrics - Use for broken alter_status command

### PR DESCRIPTION
**Pull request by virajjasani**
Jira: https://issues.apache.org/jira/browse/HBASE-22903

### CODE REVIEW

**Comment by virajjasani** [comment]
Please review @saintstack

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   2m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  prototool  |   0m  0s |  prototool was not available.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 41s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   7m  7s |  master passed  |
| :green_heart: |  compile  |   3m 12s |  master passed  |
| :green_heart: |  checkstyle  |   2m 56s |  master passed  |
| :green_heart: |  shadedjars  |   5m 57s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 50s |  master passed  |
| :blue_heart: |  spotbugs  |   5m 33s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |  10m  4s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   6m 28s |  the patch passed  |
| :green_heart: |  compile  |   3m 14s |  the patch passed  |
| :green_heart: |  cc  |   3m 14s |  the patch passed  |
| :green_heart: |  javac  |   3m 15s |  the patch passed  |
| :broken_heart: |  checkstyle  |   0m 41s |  hbase-client: The patch generated 9 new + 180 unchanged - 0 fixed = 189 total (was 180)  |
| :broken_heart: |  rubocop  |   0m 11s |  The patch generated 3 new + 270 unchanged - 1 fixed = 273 total (was 271)  |
| :green_heart: |  whitespace  |   0m  1s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   6m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  22m 10s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  hbaseprotoc  |   2m 59s |  the patch passed  |
| :green_heart: |  javadoc  |   1m 49s |  the patch passed  |
| :green_heart: |  findbugs  |  10m 31s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 53s |  hbase-protocol-shaded in the patch passed.  |
| :green_heart: |  unit  |   2m 11s |  hbase-client in the patch passed.  |
| :broken_heart: |  unit  | 290m 46s |  hbase-server in the patch failed.  |
| :green_heart: |  unit  |  11m 25s |  hbase-shell in the patch passed.  |
| :green_heart: |  asflicense  |   1m 58s |  The patch does not generate ASF License warnings.  |
|  |   | 406m 47s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestSCPWithReplicasWithoutZKCoordinated |
|   | hadoop.hbase.client.TestAsyncRegionAdminApi2 |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/611 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux d62b8f9b8295 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-611/out/precommit/personality/provided.sh |
| git revision | master / a58149827b |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/1/artifact/out/diff-checkstyle-hbase-client.txt |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/1/artifact/out/diff-patch-rubocop.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/1/testReport/ |
| Max. process+thread count | 4712 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.74.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
Please review @Reidddddd @Apache9

**Comment by virajjasani** [comment]
Please review @anoopsjohn @Reidddddd

**Comment by virajjasani** [comment]
Yes, Jira Id is correct, however, the solution is generic i.e. use new metrics.
Will update the PR title

**Comment by Reidddddd** [comment]
Good to know, i'm reviewing another big PR.
Will be back.

**Comment by virajjasani** [comment]
@Reidddddd Please review if you are available
Thanks

**Comment by virajjasani** [comment]
Requesting your review @saintstack

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 36s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  prototool  |   0m  0s |  prototool was not available.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 34s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 28s |  master passed  |
| :green_heart: |  compile  |   2m 28s |  master passed  |
| :green_heart: |  checkstyle  |   2m 25s |  master passed  |
| :green_heart: |  shadedjars  |   4m 47s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 29s |  master passed  |
| :blue_heart: |  spotbugs  |   4m 20s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   8m 11s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   5m  7s |  the patch passed  |
| :green_heart: |  compile  |   2m 30s |  the patch passed  |
| :green_heart: |  cc  |   2m 30s |  the patch passed  |
| :green_heart: |  javac  |   2m 30s |  the patch passed  |
| :broken_heart: |  checkstyle  |   0m 41s |  hbase-client: The patch generated 1 new + 180 unchanged - 0 fixed = 181 total (was 180)  |
| :broken_heart: |  rubocop  |   0m 10s |  The patch generated 3 new + 256 unchanged - 1 fixed = 259 total (was 257)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   6m  1s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  20m 17s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  hbaseprotoc  |   2m 53s |  the patch passed  |
| :green_heart: |  javadoc  |   1m 40s |  the patch passed  |
| :green_heart: |  findbugs  |  10m  7s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 51s |  hbase-protocol-shaded in the patch passed.  |
| :green_heart: |  unit  |   2m 10s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 162m 28s |  hbase-server in the patch passed.  |
| :green_heart: |  unit  |   7m 14s |  hbase-shell in the patch passed.  |
| :green_heart: |  asflicense  |   2m  8s |  The patch does not generate ASF License warnings.  |
|  |   | 260m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/611 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux 78f96f1a122f 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-611/out/precommit/personality/provided.sh |
| git revision | master / 3250a8021a |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/2/artifact/out/diff-checkstyle-hbase-client.txt |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/2/artifact/out/diff-patch-rubocop.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/2/testReport/ |
| Max. process+thread count | 4806 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.74.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
@Apache9 could you please review this PR?

File: hbase-shell/src/main/ruby/hbase/admin.rb [hunk 1]
```diff
@@ -611,16 +611,19 @@ def alter_status(table_name)
...
-      status = Pair.new
       begin
-        status = @admin.getAlterStatus(org.apache.hadoop.hbase.TableName.valueOf(table_name))
-        if status.getSecond != 0
-          puts "#{status.getSecond - status.getFirst}/#{status.getSecond} regions updated."
+        cluster_metrics = @admin.getClusterMetrics
+        table_region_status = cluster_metrics.getTableRegionStatesCount
+                                  .get(org.apache.hadoop.hbase.TableName.valueOf(table_name))
+        if table_region_status.getTotalRegions != 0
+          updated_regions = table_region_status.getTotalRegions -
```
[the remark this answers is not in the document]
Comment (virajjasani) [reply hbase-shell/src/main/ruby/hbase/admin.rb]: I believe we can minus closed region, but opening should be part of RIT already

**Comment by openinx** [comment]
BTW, please also take a look the HBaseQA, seems checkstyles & ruby has some problem now, please update your PR if it's related. @virajjasani

File: hbase-shell/src/main/ruby/hbase/admin.rb [hunk 2]
```diff
@@ -611,16 +611,19 @@ def alter_status(table_name)
       # Table should exist
       raise(ArgumentError, "Can't find a table: #{table_name}") unless exists?(table_name)
 
-      status = Pair.new
       begin
-        status = @admin.getAlterStatus(org.apache.hadoop.hbase.TableName.valueOf(table_name))
-        if status.getSecond != 0
-          puts "#{status.getSecond - status.getFirst}/#{status.getSecond} regions updated."
+        cluster_metrics = @admin.getClusterMetrics
```
[the remark this answers is not in the document]
Comment (virajjasani) [reply hbase-shell/src/main/ruby/hbase/admin.rb]: Done

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  prototool  |   0m  0s |  prototool was not available.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 33s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 45s |  master passed  |
| :green_heart: |  compile  |   2m 43s |  master passed  |
| :green_heart: |  checkstyle  |   2m 37s |  master passed  |
| :green_heart: |  shadedjars  |   5m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 26s |  master passed  |
| :blue_heart: |  spotbugs  |   4m 53s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   8m 45s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   6m 19s |  the patch passed  |
| :green_heart: |  compile  |   2m 57s |  the patch passed  |
| :green_heart: |  cc  |   2m 57s |  the patch passed  |
| :green_heart: |  javac  |   2m 57s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 45s |  the patch passed  |
| :broken_heart: |  rubocop  |   0m 16s |  The patch generated 4 new + 418 unchanged - 1 fixed = 422 total (was 419)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   5m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  19m 25s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  hbaseprotoc  |   2m 46s |  the patch passed  |
| :green_heart: |  javadoc  |   1m 33s |  the patch passed  |
| :green_heart: |  findbugs  |   9m 51s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 47s |  hbase-protocol-shaded in the patch passed.  |
| :green_heart: |  unit  |   2m  5s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 174m  8s |  hbase-server in the patch passed.  |
| :green_heart: |  unit  |   7m 24s |  hbase-shell in the patch passed.  |
| :green_heart: |  asflicense  |   1m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 273m 21s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/611 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux 7dbacadc9a87 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-611/out/precommit/personality/provided.sh |
| git revision | master / 1170f28122 |
| Default Java | 1.8.0_181 |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/3/artifact/out/diff-patch-rubocop.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/3/testReport/ |
| Max. process+thread count | 4289 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.74.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
Thanks for the review, updated the PR based on comments, please take a look @openinx 
Some rubocop warning would be difficult to resolve for any changes in ruby file like Metrics/AbcSize etc, which is usually ignored IMHO.
Thanks

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  prototool  |   0m  0s |  prototool was not available.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 44s |  master passed  |
| :green_heart: |  compile  |   2m 27s |  master passed  |
| :green_heart: |  checkstyle  |   2m 29s |  master passed  |
| :green_heart: |  shadedjars  |   4m 54s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 21s |  master passed  |
| :blue_heart: |  spotbugs  |   4m 24s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   8m 11s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   5m 28s |  the patch passed  |
| :green_heart: |  compile  |   2m 23s |  the patch passed  |
| :green_heart: |  cc  |   2m 23s |  the patch passed  |
| :green_heart: |  javac  |   2m 23s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 25s |  the patch passed  |
| :broken_heart: |  rubocop  |   0m 12s |  The patch generated 1 new + 418 unchanged - 1 fixed = 419 total (was 419)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   5m  1s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  17m  2s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  hbaseprotoc  |   2m 19s |  the patch passed  |
| :green_heart: |  javadoc  |   1m 20s |  the patch passed  |
| :green_heart: |  findbugs  |   8m 41s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 41s |  hbase-protocol-shaded in the patch passed.  |
| :green_heart: |  unit  |   1m 48s |  hbase-client in the patch passed.  |
| :green_heart: |  unit  | 156m 33s |  hbase-server in the patch passed.  |
| :green_heart: |  unit  |   7m  8s |  hbase-shell in the patch passed.  |
| :green_heart: |  asflicense  |   1m 36s |  The patch does not generate ASF License warnings.  |
|  |   | 247m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/611 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux dc7f31295b35 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-611/out/precommit/personality/provided.sh |
| git revision | master / ce0fbc2af9 |
| Default Java | 1.8.0_181 |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/4/artifact/out/diff-patch-rubocop.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/4/testReport/ |
| Max. process+thread count | 4294 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/4/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.74.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionStatesCount.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  private int totalRegions;
+
+  private RegionStatesCount() {
+  }
+
+  public int getClosedRegions() {
+    return closedRegions;
+  }
+
+  private void setClosedRegions(int closedRegions) {
```
Comment (openinx) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionStatesCount.java]: nit:  let's try to align the get & set placement ?   all getXX first, then setXX ? or  getXX & setXX one by one ?

Comment (virajjasani) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/RegionStatesCount.java]: Done @openinx 
Thanks

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :blue_heart: |  prototool  |   0m  0s |  prototool was not available.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ master Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 35s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   5m 54s |  master passed  |
| :green_heart: |  compile  |   2m 36s |  master passed  |
| :green_heart: |  checkstyle  |   2m 32s |  master passed  |
| :green_heart: |  shadedjars  |   5m  6s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 26s |  master passed  |
| :blue_heart: |  spotbugs  |   4m 44s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   8m 43s |  master passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   5m 54s |  the patch passed  |
| :green_heart: |  compile  |   3m  6s |  the patch passed  |
| :green_heart: |  cc  |   3m  5s |  the patch passed  |
| :green_heart: |  javac  |   3m  5s |  the patch passed  |
| :green_heart: |  checkstyle  |   2m 54s |  the patch passed  |
| :broken_heart: |  rubocop  |   0m 15s |  The patch generated 1 new + 418 unchanged - 1 fixed = 419 total (was 419)  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   6m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  20m 33s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| :green_heart: |  hbaseprotoc  |   2m 53s |  the patch passed  |
| :green_heart: |  javadoc  |   1m 42s |  the patch passed  |
| :green_heart: |  findbugs  |  10m 27s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 48s |  hbase-protocol-shaded in the patch passed.  |
| :green_heart: |  unit  |   2m  7s |  hbase-client in the patch passed.  |
| :broken_heart: |  unit  | 173m 31s |  hbase-server in the patch failed.  |
| :green_heart: |  unit  |   7m 16s |  hbase-shell in the patch passed.  |
| :green_heart: |  asflicense  |   1m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 275m 33s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.procedure.TestMasterFailoverWithProcedures |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/611 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile cc hbaseprotoc prototool rubocop |
| uname | Linux 958f7e29d7fd 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-611/out/precommit/personality/provided.sh |
| git revision | master / ca0d9f3cd2 |
| Default Java | 1.8.0_181 |
| rubocop | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/5/artifact/out/diff-patch-rubocop.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/5/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/5/testReport/ |
| Max. process+thread count | 4225 (vs. ulimit of 10000) |
| modules | C: hbase-protocol-shaded hbase-client hbase-server hbase-shell U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-611/5/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 rubocop=0.74.0 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**